provider/equinix/environ.go

```diff
@@ -134,7 +134,7 @@ func (e *environ) ControllerInstances(ctx context.ProviderCallContext, controlle
 	if err != nil {
 		return nil, err
 	}
-	instanceIDs := []instance.Id{}
+	instanceIDs := make([]instance.Id, len(insts))
 	for _, i := range insts {
```

Comment:
If you pre-allocate the slice, you should change your for loop as follows:

```go
for i, inst := range insts {
  instanceIDs[i] = inst.Id()
}
```

Alternatively, if you want to use append here, you can chance the slice allocation line to:

```
instanceIDs := make([]instance.Id, 0, len(insts))
```